compiler/passes/scopeResolve.cpp

```diff
@@ -1600,12 +1640,14 @@ static void printConflictingSymbols(std::vector<Symbol*>& symbols, Symbol* sym)
 // by that name in the context of that call
 Symbol* lookupAndCount(const char*           name,
                        BaseAST*              context,
-                       int&                  nSymbolsFound) {
+                       int&                  nSymbolsFound,
+                       SymRename**           rename) {
 
   std::vector<Symbol*> symbols;
+  std::map<Symbol*, SymRename*> renames;
```

Comment:
Is this map only used in error reporting? If so, we might want to streamline the non-error case (so it doesn't include the map) and then do additional (slower) work if an error occurred to get a good message.

Reply:
That sounds plausible to me

Reply:
Unfortunately, waiting for error detection would mean the cases I mentioned in the other thread would be ignored, so I'm going to handle this with that thread